src/chrony.py

```diff
@@ -261,13 +261,13 @@ def _make_certs_dir(self) -> None:  # pragma: nocover
         self.CERTS_DIR.mkdir(exist_ok=True, mode=0o700)
         shutil.chown(self.CERTS_DIR, "_chrony", "_chrony")
 
-    def _iter_certs_dir(self) -> typing.Iterator[pathlib.Path]:
+    def _iter_certs_dir(self) -> list[pathlib.Path]:  # pragma: nocover
```

Comment:
I think there should be an explanatory note at the top explaining why these lines are not covered.

Reply:
There are filesystem operations or subprocess commands that can't be performed in unit tests (that's why they are separated into special functions, which makes them easier to patch). There are just too many of them (a total of 11), and they all exist for the same reason.

Reply:
Would you prefer that I add them? If so, I can do that.

Reply:
can't you mockpatch `CERTS_DIR` using the tmp_path fixture and then run the code in unit tests?

Reply:
When writing to the `CERTS_DIR`, the file ownership must be `_chrony`. This means that on systems without the `_chrony` user and group, it will fail. Also, when using `tempfile`, the full filename becomes unpredictable, causing the generated configuration file to be unpredictable as well, which can cause some problem.